compiler/rustc_middle/src/query/mod.rs

```diff
@@ -2124,4 +2124,9 @@ rustc_queries! {
     ) -> Result<(), ErrorGuaranteed> {
         desc { |tcx| "checking assoc const `{}` has the same type as trait item", tcx.def_path_str(key.0.to_def_id()) }
     }
+
+    query deduced_param_attrs(def_id: DefId) -> &'tcx [ty::DeducedParamAttrs] {
+        desc { |tcx| "deducing parameter attributes for {}", tcx.def_path_str(def_id) }
+        separate_provide_extern
```

Comment:
Adding `cache_on_disk_if { def_id.is_local() }` may help in the incremental case.
(Exact syntax to be verified based on what other queries do.)